### PR TITLE
openocd: esp32: add debugging support

### PR DESCRIPTION
**Pull request by sylvioalves**

### CODE REVIEW

File: scripts/west_commands/runners/openocd.py [hunk 1]
```diff
@@ -66,8 +67,12 @@ def __init__(self, cfg, pre_init=None, reset_halt_cmd=DEFAULT_OPENOCD_RESET_HALT
         self.gdb_cmd = [cfg.gdb] if cfg.gdb else None
         self.tui_arg = ['-tui'] if tui else []
         self.halt_arg = [] if no_halt else ['-c halt']
+        self.no_init = [] if no_init else ['-c init']
+        self.no_targets = [] if no_targets else ['-c targets']
         self.serial = ['-c set _ZEPHYR_BOARD_SERIAL ' + serial] if serial else []
         self.use_elf = use_elf
+        self.gdb_init = gdb_init
+        self.no_load = [] if no_load else ['-ex', 'load']
```
Comment (dcpleung) [inline scripts/west_commands/runners/openocd.py:75]: Could you remove the `no_` prefix for these variables? Just like `halt_arg` above. How these are being used is actually the opposite of what the names suggest.

**Review by dcpleung — CHANGES_REQUESTED** [review]

**Review by uLipe — APPROVED** [review]

File: scripts/west_commands/runners/openocd.py (hunk 1 above)
Comment (sylvioalves) [reply scripts/west_commands/runners/openocd.py]: @dcpleung I updated it to folllow `no_halt` argument and `halt_args` variable.

**Review by dcpleung — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]
I'm fine with this going in, as debug support is quite critical and no code paths are touched, just support. 
Pending the outcome of #38540

**Comment by sylvioalves** [comment]
> I'm fine with this going in, as debug support is quite critical and no code paths are touched, just support.
> Pending the outcome of #38540

This PR also changes `openocd.py`. Have you checked as well?

**Comment by cfriedt** [comment]
> > I'm fine with this going in, as debug support is quite critical and no code paths are touched, just support.
> > Pending the outcome of #38540
> 
> This PR also changes `openocd.py`. Have you checked as well?

attn: @mbolivar-nordic

File: boards/xtensa/esp32s2_saola/board.cmake [hunk 2]
```diff
@@ -1,3 +1,17 @@
 # SPDX-License-Identifier: Apache-2.0
 
+board_runner_args(openocd  --cmd-pre-init "set ESP_RTOS none")
+board_runner_args(openocd  --cmd-pre-init "set ESP32_ONLYCPU 1")
+
+# select FT2232 as default interface (as in ESP-WROVER-KIT or ESP-Prog)
+board_runner_args(openocd  --config "interface/ftdi/esp32s2_kaluga_v1.cfg")
+board_runner_args(openocd  --config "target/esp32s2.cfg")
+
+set (ESPRESSIF_OPENOCD_TOOL $ENV{ESPRESSIF_TOOLCHAIN_PATH}/openocd-esp32/bin/openocd)
```
Comment (tejlmand) [inline boards/xtensa/esp32s2_saola/board.cmake:10]: you add the espressif toolchain path to the CMakeCache here:
https://github.com/zephyrproject-rtos/zephyr/blob/0324997d1e7277374f7fcc734455eff55b7eeea1/cmake/toolchain/espressif/generic.cmake#L4

so why use the environment setting here ?

**Comment by tejlmand** [comment]
would like @mbolivar-nordic to take a look at the runners.

File: boards/xtensa/esp32s2_saola/board.cmake (hunk 2 above)
Comment (sylvioalves) [reply boards/xtensa/esp32s2_saola/board.cmake]: Not necessary. Fixed.

**Review by tejlmand — DISMISSED** [review]
Approved on my side, but waiting for @mbolivar-nordic to give his comments.

**Comment by sylvioalves** [comment]
@cfriedt Can you re-trigger buildkite? Current error is not PR specific.

**Comment by sylvioalves** [comment]
@mbolivar-nordic, PTAL, thanks.

**Comment by cfriedt** [comment]
@carlescufi - do you have any feedback for @sylvioalves while we are waiting for @mbolivar-nordic ?

**Review by carlescufi — APPROVED** [review]
LGTM, merging this without @mbolivar-nordic's review.

**Pull request merged**